misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -180,4 +234,28 @@ data class FakeJob(
   override fun deadLetter() {
     deadLettered = true
   }
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
I’m anxious about any object where the equals() state is different from the actual state. For example, tests would pass if you assertEquals() on two jobs with the same ID and different bodies.

Rollback this part of the change?

Reply:
The reason I added this override are two vars:
```
  var acknowledged: Boolean = false
  var deadLettered: Boolean = false
```
FakeJob is used in `queue.remove()` so this is important. Well, we basically don't use FakeJobQueue in tests with concurrency involved, so I'd not care, but misk is a very shared library and it uses ConcurrentHashMap etc., so to stay multi-thread safe, we need this override here.

But I can include more immutable fields: all of vals. Would that work?

Reply:
So equals and hashCode become:
```


  override fun equals(other: Any?): Boolean {
    return this === other
  }

  override fun hashCode(): Int {
    return System.identityHashCode(this)
  }
```
Does this look good?

Reply:
Or should I better make FakeJob not a data class: `class FakeJob(` and then skip overriding hashCode/equals?

Reply:
(Dropped my previous comments: nevermind)
The only place where it was critical is  when we `find and remove`, and this would be the same instance. So even if FakeJob was concurrently modified (it has two vars), the instance would be the same.. I have just overthought this.. So Im rolling this back.